packages/code-snippet/src/CodeSnippetWidget.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+
+import { IDocumentManager } from '@jupyterlab/docmanager';
+import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
+import '../style/index.css';
+import { CodeSnippetManager, ICodeSnippet } from './CodeSnippet';
+import { ReactWidget } from '@jupyterlab/apputils';
+import { ExpandableComponent } from './ExpandableComponent';
+
+/**
+ * The CSS class added to code snippet widget.
+ */
+const CODE_SNIPPETS_CLASS = 'elyra-CodeSnippets';
+const CODE_SNIPPETS_HEADER_CLASS = 'elyra-codeSnippetsHeader';
+const CODE_SNIPPET_ITEM = 'elyra-codeSnippet-item';
+const BUTTON_CLASS = 'elyra-button';
+const COPY_ICON_CLASS = 'elyra-copy-icon';
+const INSERT_ICON_CLASS = 'elyra-add-icon';
+const CODE_SNIPPET_NAME_CLASS = 'elyra-codeSnippet-name';
+const CODE_SNIPPET_BUTTONS_WRAPPER_CLASS = 'elyra-codeSnippet-buttons';
+
+/**
+ * A widget for code-snippet.
+ */
+
+class CodeSnippetTable extends React.Component<{}, any> {
+  constructor(props: any) {
+    super(props);
+    this.state = { codeSnippets: [] };
+  }
+
+  async fetchData(): Promise<ICodeSnippet[]> {
+    const codeSnippetManager = new CodeSnippetManager();
+    const codeSnippets: ICodeSnippet[] = await codeSnippetManager.findAll();
+    return codeSnippets;
+  }
+
+  // TODO: Use code mirror to display code
+  // TODO: implement copy to clipboard command
+  // TODO: implement insert code to file editor command (first check for code language matches file editor kernel language)
+  renderCodeSnippetsDisplay(): Array<JSX.Element> {
+    const codeSnippetDisplayList: Array<JSX.Element> = this.state.codeSnippets.map(
+      (codeSnippet: any, index: number) => {
+        const displayName =
+          '[' + codeSnippet.language + '] ' + codeSnippet.displayName;
```

Comment:
Will these be ordered by language? (Or can they be?)  This doesn't need to happen now, but when there are lots of snippets across multiple languages, this ordering would be helpful.  (Then order by displayName within each language.)

Reply:
We were thinking about ordering, but also a way to select only one language.

Reply:
:smile: https://github.com/elyra-ai/elyra/pull/241#discussion_r401272154